### ISSUE TITLE
`@deprecated()` stopped working since 1.1.364

### ISSUE THREAD

**Issue by InSyncWithFoo** [microsoft/pyright#8161]
Minimal reproducible example (playground links: [1.1.363](https://pyright-play.net/?pyrightVersion=1.1.363&pythonVersion=3.13&strict=true&code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigBMsEQsBjAQ3zoFgAoXlgDZtChKAGEAXL15RZUAAIMmrDljoAKAOQBZLDAAWYOpoCUMuQ2BRg6wlgHATUALQA%2BKADlSWKTzn%2BoADpg6R4xdRNAmycAuQBiKCwQcBAJKAAVfSwoCD1DOigAImBC8hQoQWFRQrFSpFElZnZOXiA), [1.1.367](https://pyright-play.net/?pyrightVersion=1.1.367&pythonVersion=3.13&strict=true&code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigBMsEQsBjAQ3zoFgAoXlgDZtChKAGEAXL15RZUAAIMmrDljoAKAOQBZLDAAWYOpoCUMuQ2BRg6wlgHATUALQA%2BKADlSWKTzn%2BoADpg6R4xdRNAmycAuQBiTzADVDReIA)):

```python
from typing_extensions import deprecated

class C:
    @deprecated('Method')
    def f(self) -> None:
        ...
```

```python
C().f()  # 1.1.367: no errors
         # 1.1.363: The method "f" in class "C" is deprecated
```

**Comment by erictraut** [comment]
This was due to a recent change in the typeshed stubs. It specifically affects Python 3.13, not earlier versions. It is already fixed in pyright's "main" branch and will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.368.
